### DISCUSSION TITLE
Need more in depth documentation about caching mechanism

### DISCUSSION THREAD

**Discussion by modbender** [johannschopplich/nuxt-api-party#15] [Ideas]
I think the current documentation doesn't cover all aspect of information regarding cache. I need in depth explanation about caching mechanism. My website updates almost every hour (private site), and it could be crucial if there's a update that's not rendered on time of change.
So what I need is:

1. In depth documentation of cache as much as possible, like when it will update, or how when to update is determined.
2. Manual timeout option both composable argument level and global.
3. Global options to disable/enable cache.

**Comment by johannschopplich** [comment]
Great ideas. I want to set up a documentation for Nuxt API Party soon. I'd consider this topic to get a separate doc page.

**Reply by modbender** [reply]
Btw unrelated, and just my opinion, but I think for creating documentation you can check out Next.js powered Nextra

https://nextra.site/

Totally automatic, you just have to write MD and it creates everything else (Search bar and stuff). You can deploy for free on Vercel.

**Reply by johannschopplich** [reply]
Great idea, but I'd rather stick to the Vue ecosystem. Although most Nuxt modules stick to [Docus](https://docus.dev/) (created by the Nuxt team), I do like VitePress much more. 

Initial version of the docs are ready. In the next phase I will add a page dedicated to caching. 

👉 [nuxt-api-party.jhnn.dev](https://nuxt-api-party.jhnn.dev/)

**Reply by johannschopplich** [reply]
@modbender The first version of the caching page is ready: https://nuxt-api-party.jhnn.dev/guide/caching.html

**Reply by modbender** [reply]
Did you rename `usePartyData` to `useApiPartyData`?
Some parts of the page say it's usePartyData and some not:
https://nuxt-api-party.jhnn.dev/api/use-api-party-data.html

**Reply by johannschopplich** [reply]
Excellent catch! Thanks, updated: https://github.com/johannschopplich/nuxt-api-party/commit/80fb37df171106487cb53d638950256e40525e41

**Reply by modbender** [reply]
@johannschopplich Thanks for caching documentation. I have a basic understanding of it now.
Btw I have a question, is the cache authentication friendly. Talking about header `Authorization`.

EDIT: You can forget about the other thing I asked here.

**Reply by johannschopplich** [reply]
> Btw I have a question, is the cache authentication friendly. Talking about header `Authorization`.

What do you mean by that? Every option of both generated composables are cache-friendly. Technically speaking, the header is not part of the calculated cache hash:

https://github.com/johannschopplich/nuxt-api-party/blob/e72dbff198bdfcd36f77656985726d57ba7e80c7/src/runtime/composables/useApiData.ts#L103

If you intended to have a reactive header, you can add a reactive variable to the `watch` option of your `useApiPartyData` composable.

**Comment by modbender** [comment]
@johannschopplich Is it possible for you to implement something like stale-while-revalidate?
Basically what I mean is, the response with unique key will be cached until a timeout, and after timeout, the fresh data will be fetched in background (non blocking) and until this fresh data is fetched, cached data will be served.

**Reply by johannschopplich** [reply]
This is a feature request. Could you please open a new discussion? Thanks!

**Reply by modbender** [reply]
@johannschopplich Done https://github.com/johannschopplich/nuxt-api-party/discussions/18
